### PR TITLE
PI-8297: Product images removed after returning from reviews page

### PR DESCRIPTION
**Pull request by CarinaHoffmann**
- implement fallback for the case that a child product doesn't have images
- add selector to get current variant id
- add conditions to setting and unset variant/base product ids

### CODE REVIEW

File: product/subscriptions/index.js [hunk 1]
```diff
@@ -26,13 +28,29 @@ function product(subscribe) {
...
+    }
+
+    dispatch(setProductId(baseProductId));
   });
 
-  subscribe(itemRouteDidLeave$, ({ dispatch }) => {
-    dispatch(setProductId(null));
+  subscribe(itemRouteDidLeave$, ({ dispatch, getState }) => {
+    const state = getState();
+    /*
```
Comment (richardgorman) [inline product/subscriptions/index.js]: Please stick to the usual multiline comment style.

File: product/subscriptions/index.js [hunk 2]
```diff
@@ -26,13 +28,29 @@ function product(subscribe) {
...
+     If there is a redirect to login we don't want the state to be mutated,
```
Comment (richardgorman) [inline product/subscriptions/index.js]: This comment should be more specific with what it describes.

File: product/subscriptions/index.js [hunk 3]
```diff
@@ -26,13 +28,29 @@ function product(subscribe) {
     !pathname.startsWith(ITEM_PATH)
   ));
 
-  subscribe(itemRouteDidEnter$, ({ dispatch, pathname }) => {
-    const productId = hex2bin(pathname.split('/')[2]);
-    dispatch(setProductId(productId));
+  subscribe(itemRouteDidEnter$, ({ dispatch, getState }) => {
+    const state = getState();
+    const baseProductId = hex2bin(state.history.pathname.split('/')[2]);
```
Comment (richardgorman) [inline product/subscriptions/index.js]: This could become a selector.

**Comment by richardgorman** [comment]
The fix seems to work. However, it's not good to have actions fire that result in no change. I see that when I navigate to the gallery route that the `SET_PRODUCT_ID` (for example) is fired with the current product id. There should be a check to not trigger this action if the value is not changing.

File: product/subscriptions/index.js (hunk 1 above)
Comment (CarinaHoffmann) [reply product/subscriptions/index.js]: Done.

File: product/subscriptions/index.js (hunk 2 above)
Comment (CarinaHoffmann) [reply product/subscriptions/index.js]: Done.

File: product/subscriptions/index.js (hunk 3 above)
Comment (CarinaHoffmann) [reply product/subscriptions/index.js]: Done.

**Comment by CarinaHoffmann** [comment]
Done.

File: product/selectors/product.js [hunk 4]
```diff
@@ -20,13 +22,31 @@ import { getCurrentCategoryId } from '../../category/selectors';
...
  */
 export const getCurrentProductId = state =>
   state.product.currentProduct.productVariantId || state.product.currentProduct.productId;
 
+/**
+ * Gets the product id from the current history state pathname.
+ * @param {Object} state The current application state.
+ * @return {null|string} Product id or null
+ */
+export const getHistoryPathProductId = (state) => {
```
Comment (richardgorman) [inline product/selectors/product.js]: You wanted to create a selector here but this is not a selector. It's just a function that returns something.

**Comment by richardgorman** [comment]
The check that prevents the product id being set unnecessary is still missing.

File: product/selectors/product.js (hunk 4 above)
Comment (CarinaHoffmann) [reply product/selectors/product.js]: done

**Comment by CarinaHoffmann** [comment]
> The check that prevents the product id being set unnecessary is still missing.

Done.

> Missing selector intention

Done.

**Pull request merged**